.github/workflows/fetch-lib.yml

```
@@ -0,0 +1,40 @@
+name: Update Dependencies (fetch-lib)
+
+on:
+  schedule:
+    - cron:  '0 10 * * 1'
+  workflow_call:
+    secrets:
+      CHARMHUB_TOKEN:
+        required: true
+
+jobs:
+  fetch-lib:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup operator environment
+        uses: charmed-kubernetes/actions-operator@main
+      - name: Checkout
+        uses: actions/checkout@v3
+      - name: Check libs
+        uses: canonical/charming-actions/check-libraries@2.0.0
+        with:
+          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          use-labels: false
+      - name: Check diff
+        id: check
+        run: git diff --exit-code
+      - name: Create Pull Request
+        if: steps.check.outputs.exit-code != 0
+        uses: peter-evans/create-pull-request@v4
+        with:
+            commit-message: Update dependencies (fetch-lib)
+            title: Update dependencies (fetch-lib)
+            body: |
+              - Dependency updates
+  
+              Auto-generated by [create-pull-request][1]
+  
+              [1]: https://github.com/peter-evans/create-pull-request
+            branch: fetch-lib
```

As is, this action is allowed to create PRs but these PRs will not trigger the repo's CI.  This is because the action uses the `GITHUB_TOKEN` to create the pull request, but that token does not have permission to trigger workflow events.  There's more documentation [here].

If you want to trigger the repo's CI (which I'm guessing you do), you need to create and pass a PAT that has `repo` access on the repo where the PR is being created.  So something like:

```suggestion
            branch: fetch-lib
            token: "${{ secrets.SOME_PAT_TOKEN }}"
```

---

@jnsgruk that looks great. I'm not clear on the `tibdex/github-app-token` details but would like to give it a try.

---

Could this be part of template-operator?